### PR TITLE
basic external includes for js2dt

### LINKED ISSUE #58: support for remote $ref in js2dt
E.g.
```
{
    "$schema": "http://json-schema.org/draft-06/schema#",
    "description": "JSON Schema with a remote $ref",
    "type": "object",
    "properties": {
        "geo": { "$ref": "http://json-schema.org/geo" }
    }
}
```
should fetch the JSON Schema referenced in $ref

### PR DESCRIPTION
**Pull request by ForsakenHarmony**
closes #58

### CODE REVIEW

**Comment by jstoiko** [comment]
Do we want to add an integration test that has a `$ref` pointing to a remote URL since integration tests are only either ran manually or ran by Travis CI. In both case having the ability to run offline wouldn't be a big issue...

File: src/js2dt.js [hunk 1]
```diff
@@ -404,9 +411,18 @@ function convertPatternProperties (data) {
...
-  data['type'] = utils.typeNameFromRef(data['$ref'])
+  let loc = data['$ref']
+  let split = loc.split('#')
   delete data['$ref']
-  return data
+
+  if (loc[0] === '#') {
+    data['type'] = utils.typeNameFromRef(loc)
+    return data
+  } else if (loc.slice(0, 4) === 'http') {
```
Comment (jstoiko) [inline src/js2dt.js]: we should also support `https`

**Comment by jstoiko** [comment]
if I run `js2dt` against this schema: http://json-schema.org/example/calendar.json, the value of the remote $ref `geo` doesn't get properly translated. I.e.:
```
$ js2dt calendar.json 
#%RAML 1.0 DataType
type: object
properties:
  (...)
  geo?: |-
    {
        "id": "http://json-schema.org/geo",
        "$schema": "http://json-schema.org/draft-06/schema#",
        "description": "A geographical coordinate",
        "type": "object",
        "properties": {
            "latitude": { "type": "number" },
            "longitude": { "type": "number" }
        }
    }
```

File: src/js2dt.js (hunk 1 above)
Comment (ForsakenHarmony) [reply src/js2dt.js]: well https also begins with http

Comment (jstoiko) [reply src/js2dt.js]: 🙃

**Comment by ForsakenHarmony** [comment]
fixed now

**Pull request merged**